### ISSUE TITLE
Wrapper script

### ISSUE THREAD

**Issue by dlesl** [borkdude/deps.clj#81]
Thanks for this tool, I think it has a lot of potential to reduce friction especially in CI environments, as you mention in the README. I'm wondering if it would make sense to take that even further by providing a wrapper script designed to be checked into a repo, like gradle does with its wrapper. Then, after checkout, all one would need to do is run eg.

    $ ./deps -A:test

and java would be the only prerequisite. [Here](https://github.com/dlesl/deps-wrapper)'s a small POC to demonstrate how it could work

**Comment by borkdude** [comment]
In theory the idea is good, but I wouldn't use the standalone jar for this since you then start a nested JVM each time you start Clojure, which is slower than necessary. The deps tool should only take a few milliseconds and that tool then launches the JVM (with Clojure dependencies). 

Note that there is also a fully standalone cross platform `deps.clj` script you can download from the root of the repo which you can run with [babashka](https://babashka.org/) for fast startup.

**Comment by dlesl** [comment]
Makes sense, I guess a more advanced script could download the appropriate native binary and use that? Actually it could just be a version of `install` that execs itself at the end?

The thing I'm really after is a way (any way, even if a bit slow) to get into a clojure environment without installing anything first. I'm thinking of two use cases, internal tools etc where you want to be able to distribute source, but still be able to say "run this command" and it just works (installing something is somehow a bigger psychological barrier to cross). The second is CI, where your organisation probably already has a runner with Java installed.
